airbyte-migration/src/main/java/io/airbyte/migrate/MigrateWithMetadata.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.migrate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.commons.functional.Consumers;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * Additionally there are some migration metadata fields that are important to fill in on each
+ * migration. Instead of leaving this up to the person writing the migration, this class handles it
+ * for them. It writes a record saying when the migration happens. It also updates the version in
+ * the AIRBYTE_METADATA table.
+ */
+public class MigrateWithMetadata {
```

Comment:
This class looks like a decorator for a migration, what about it extends Migration?

Reply:
Migrations.addMetadataToMigration(migration)

Reply:
Yeah. It started out as one. Then I felt like I was stubbing out a bunch of methods for no real reason. Additionally, I want to make it super clear that version metadata is handled by the app, not a responsibility of the migration. We will see how it evolves. We can definitely turn it back into one but for now this is less code and a clearer delineation of the responsibilities of a migration versus the app.